Lesson18/sahra/script.js

```diff
@@ -0,0 +1,170 @@
+// 1. Write a function that takes an array of integers and returns the sum of all elements.
+// Input: [1, 2, 3, 4]
+// Output: 10
+const integersEx1 = [1, 2, 3, 4];
+const integersEx1V2 = [10, 2, 30, 4];
+function sum(myArr){
+  let sum = 0;
+  for(let i = 0; i < myArr.length; i++){
+    sum += myArr[i];
+  }
+  return sum;
+}
+const result = sum(integersEx1V2);
+console.log(result);
+
+console.log('----------------------------');
+// 2. Write a function that takes an array of integers and returns the average of all elements.
+//Input: [1, 2, 3, 4]
+// Output: 2.5
+function numbers(myArr){
+  let sum = 0;
+  for(let i=0; i < myArr.length; i++){
+    sum += myArr[i];
+  }
+  sum /= myArr.length;
+  return sum;
+}
+const example2 = [4,8,12,20];
+const resultEx2 = numbers(example2);
+console.log(resultEx2);
+
+console.log('----------------------------');
+// 3. Write a function that takes an array of strings and concatenates them together adding spaces.
+// Input: ["Hello", "World", "!"]
+// Output: "Hello World !"
+
+let stringE3 = ["Hello", "World", "!"];
+function sentence(myArr){
+  return myArr.join(" ");
+}
+const resultEx3 = sentence(stringE3);
+console.log(resultEx3);
+
+console.log('----------------------------');
+// 4. Write a function that takes an array of integers and returns a new array with odd numbers only.
+// Input: [1, 2, 3, 4, 7, 10]
+// Output: [1, 3, 7]
+let arrayEx = [1, 2, 3, 4, 7, 10];
+function odd(myArr){  
+  let numbers = []
+  for(let i=0; i < myArr.length; i++){
+  if(myArr[i] % 2 !== 0){
+    numbers.push(myArr[i]);
+  }
+  }
+  return numbers;
+}
+const resultEx4 = odd(arrayEx);
+console.log(resultEx4);
+
+
+console.log('----------------------------');
+// 5. Write a function that takes an array of integers and returns a new array with even numbers only.
+// Input: [1, 2, 3, 4, 7, 10]
+// Output: [1, 2, 4, 10] // 1 is not even
+const testArrayEx5 = [1, 2, 3, 4, 7, 10];
+function even(myArr){  
+  let numbers = []
+  for(let i=0; i < myArr.length; i++){
+  if(myArr[i] % 2 === 0){
+    numbers.push(myArr[i]);
+  }
+  }
+  return numbers;
+}
+const resultEx5 = even(testArrayEx5);
+console.log(resultEx5);
+
+console.log('----------------------------');
+// 6. Write a function that takes an array and a value, and returns a new array with all occurrences of the value removed.
+// Input: ([1, 2, 3, 4, 2], 2)
+// Output: [1, 3, 4]
+const testArrayEx6 = [1, 2, 3, 4, 2];
+function value(myArr){  
```

Comment:
```suggestion
function removeFromArray(array, valueToRemove){  
```

Rename the rest as well as function name and variable names change.